File: gommo-backend/src/main/java/br/com/gommo/modules/offboarding/mapper/OffboardingMapper.java
package br.com.gommo.modules.offboarding.mapper;
import br.com.gommo.modules.offboarding.dto.*; import br.com.gommo.modules.offboarding.entity.Offboarding; import br.com.gommo.modules.offboarding.entity.DismissalTypeEnum;
import org.springframework.stereotype.Component;
@Component public class OffboardingMapper {
    public Offboarding toEntity(OffboardingRequestDto dto) {
        return Offboarding.builder().collaboratorId(dto.getCollaboratorId()).dismissalDate(dto.getDismissalDate())
            .dismissalType(dto.getDismissalType() != null ? dto.getDismissalType() : DismissalTypeEnum.WITHOUT_CAUSE)
            .dismissalNotes(dto.getDismissalNotes()).homologationNotes(dto.getHomologationNotes()).build();
    }
    public void updateEntity(Offboarding entity, OffboardingRequestDto dto) {
        entity.setCollaboratorId(dto.getCollaboratorId()); entity.setDismissalDate(dto.getDismissalDate());
        if (dto.getDismissalType() != null) entity.setDismissalType(dto.getDismissalType());
        entity.setDismissalNotes(dto.getDismissalNotes()); entity.setHomologationNotes(dto.getHomologationNotes());
    }
    public OffboardingResponseDto toResponse(Offboarding entity) {
        return OffboardingResponseDto.builder().id(entity.getId()).status(entity.getStatus())
            .collaboratorId(entity.getCollaboratorId()).dismissalDate(entity.getDismissalDate()).dismissalType(entity.getDismissalType())
            .dismissalNotes(entity.getDismissalNotes()).homologationNotes(entity.getHomologationNotes())
            .createdAt(entity.getCreatedAt()).updatedAt(entity.getUpdatedAt()).build();
    }
}

File: gommo-admin-frontend/src/modules/offboarding/lib/offboarding.mapper.ts
import type { Offboarding, OffboardingCreateDto } from "@/modules/offboarding/dto/offboarding.dto";

export function offboardingToFormDto(entity: Offboarding): OffboardingCreateDto {
    return {
        collaboratorId: entity.collaboratorId,
        dismissalDate: entity.dismissalDate?.slice(0, 10) ?? "",
        dismissalType: entity.dismissalType,
        dismissalNotes: entity.dismissalNotes,
        homologationNotes: entity.homologationNotes,
    };
}

export const emptyOffboardingForm = (): OffboardingCreateDto => ({
    collaboratorId: "",
    dismissalDate: "",
});
